docs/concepts/internationalization.md

```diff
@@ -37,15 +37,24 @@ By default, `#getDescriptionId` will return `block.` or `item.` prepended to the
 The only purpose of a translation key is internationalization. Do not use them for logic. Use registry names instead.
 :::
 
-Localization methods
+`Component` and `ComponentContents`
 --------------------
 
+Localizable text can be represented as a tree, each node being `Component` and their text-representible contents being `ComponentContents`. `Component`'s children can be retrieved with `Component#getSiblings` (yes we are aware of weird name). When displaying `Component` to user, its own `ComponentContents` and children's are appended into one String in this order. Each `Component` has `Style`, which describes how the text component should look, such as color, bold, italic, and each children inherits parent's style.
```

Comment:
This paragraph doesn't make a lot of sense. While you can think of components as a tree, it doesn't provide any value to the consumer of the documentation as most component implementations are typically single use or chainable. Additionally, the main reason to append siblings is because of the limitations in styles, and even then you would just pass in a separate component as an object to the localizer.

`Component`s are messages that can be formatted. The contents of a component is stored within a `ComponentContent`s. The rest of the information should be separate paragraphs to understand them in more detail.

Reply:
I see. I thought it was tree as usages of `#getSiblings` are recursive loop.

Reply:
I never said they aren't. I just don't understand how that would help the consumer use it. If you're appending siblings haphazardly, the implementation is likely not well defined. It's more useful to give an overview and provide use cases instead of meticulously defining what something is. That is the point of the javadocs.